pkg/apis/extensions/v1beta1/types.go

```diff
@@ -286,6 +290,43 @@ type DeploymentStatus struct {
 
 	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
 	UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`
+
+	// Conditions represent the latest available observations of a deployment's current state.
+	Conditions []DeploymentCondition `json:"conditions,omitempty"`
+}
+
+type DeploymentConditionType string
+
+// These are valid conditions of a deployment.
+const (
+	// DeploymentProgressing means the deployment is progressing.
+	DeploymentProgressing DeploymentConditionType = "Progressing"
+	// DeploymentFailed means the deployment has failed its execution.
```

Comment:
Agreed. I need to think about how to express this as cheap as it can get. My intention is to keep the strategy as separated as it can be from any checks for inert, failed, etc. In the case of failed though we need to look for the newest rc in order to realize if the deployment hash is new which complicates things...